lib/transmuxer.js

```diff
@@ -1029,61 +1142,27 @@ Transmuxer = function() {
   this.metadataStream = new muxjs.mp2t.MetadataStream();
 
   // assemble MPEG2-TS packets into elementary streams
-  packetStream.pipe(parseStream);
-  parseStream.pipe(elementaryStream);
+  packetStream
+    .pipe(parseStream)
+    .pipe(elementaryStream);
+
   // remux the streams
   elementaryStream.pipe(aacStream);
   elementaryStream.pipe(h264Stream);
   elementaryStream.pipe(this.metadataStream);
 
-  // handle incoming data events
-  h264Stream.on('data', function(data) {
-    // record the track config
-    if (data.nalUnitType === 'seq_parameter_set_rbsp' &&
-        !config) {
-      config = data.config;
-
-      videoTrack.width = config.width;
-      videoTrack.height = config.height;
-      videoTrack.sps = [data.data];
-      videoTrack.profileIdc = config.profileIdc;
-      videoTrack.levelIdc = config.levelIdc;
-      videoTrack.profileCompatibility = config.profileCompatibility;
-    }
-    if (data.nalUnitType === 'pic_parameter_set_rbsp' &&
-        !pps) {
-      pps = data.data;
-      videoTrack.pps = [data.data];
-    }
-  });
-  // generate an init segment based on the first audio sample
-  aacStream.on('data', function(data) {
```

Comment:
This code has moved into the `push` function for the `AudioStream` constructor above. I saw no good reason for it to be here cluttering up the `Transmuxer` code.